arangod/Aql/AqlCall.cpp

```diff
@@ -26,28 +26,31 @@
 #include "Basics/Exceptions.h"
 #include "Basics/StaticStrings.h"
 #include "Basics/voc-errors.h"
+#include "Containers/FlatHashMap.h"
 #include "Logger/LogMacros.h"
 #include "Logger/Logger.h"
 
-#include <velocypack/Collection.h>
+#include <absl/strings/str_cat.h>
+#include <velocypack/Builder.h>
+#include <velocypack/Iterator.h>
 #include <velocypack/Slice.h>
 
 #include <iostream>
-#include <map>
 #include <string_view>
 
 using namespace arangodb;
 using namespace arangodb::aql;
 
 auto AqlCall::fromVelocyPack(velocypack::Slice slice) -> ResultT<AqlCall> {
   if (ADB_UNLIKELY(!slice.isObject())) {
-    using namespace std::string_literals;
-    return Result(TRI_ERROR_TYPE_ERROR,
-                  "When deserializating AqlCall: Expected object, got "s +
-                      slice.typeName());
+    return Result(
+        TRI_ERROR_TYPE_ERROR,
+        absl::StrCat("When deserializing AqlCall: Expected object, got ",
+                     slice.typeName()));
   }
 
-  auto expectedPropertiesFound = std::map<std::string_view, bool>{};
+  auto expectedPropertiesFound =
+      containers::FlatHashMap<std::string_view, bool>{};
```

Comment:
Do we seriously need a FlatHashMap for this? This map only consists of _four_ fixed entries. At this size it is probably better to use an array and do a linear search. This not only avoids any and all allocations, but I know from a past investigation that the FlatHashMap produces _an awful lot of code_ for seemingly simple methods.

Reply:
Done